### PR TITLE
Add jest/ts-jest testing framework

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

> I wanted to add a test for the promo behavior logic and I chatted with @macfarlandian about adding jest. Took a shot at it today and ran into some issues when I tried to use `react-testing-library` to render and test components. I'll try to pick this back up tomorrow to sort it out, but would love to pair on it if anyone is interested!

Some of the issues I saw that I think are related to react-testing-library: 
- All components and child components must have React explicitly imported
- For some reason I needed to import react as `import * as React from "react"` in the test and component files

I think it might be worthwhile to check out enzyme too and see if it has less issues. 

There's also a few packages I installed from trying to debug this that I'm not 100% sure we need: 
- @babel/preset-react
- jest-styled-components
- @testing-library/jest-dom

I'm also not super familiar with `parcel`, so for now I started with a mock file for svg/css assets in `src/__mocks__/`. 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes [#XXXX]

### Development

These boxes should be checked by the submitter prior to merging:

- [ ] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by macfarlandian** [comment]
@daschi I was able to at least figure out the global React issue in the commits I just pushed ... similar to what we do in the browser, I had to make a global React variable available by adding a Jest setup file. (The test in here still doesn't pass but it at least doesn't blow up without importing React.)

**Comment by szhu** [comment]
Hi, let me know here or on Slack if you want to pair!

**Comment by daschi** [comment]
@szhu happy to pair on this today if you think it still makes sense to move forward with it given your open issue! I am available anytime after 1:30pm today.

**Comment by macfarlandian** [comment]
@daschi I got your test to pass in the latest batch of commits I just pushed ... I also moved your Jest plugin imports to the new global setup file we have now so that we don't have to keep importing them into every test.

I still don't know how much the new build system will affect this configuration so I will leave it to you and @szhu to decide whether it's worthwhile to merge this first or wait until after that new system is in place. At the very least, in anticipation of that I didn't set these tests to run as Github CI checks, but we could start doing that whenever.

**Comment by daschi** [comment]
@macfarlandian That's awesome thank you! I would think we could merge this now and then adjust the configuration once the build system is updated. @szhu what do you think?

Also - while we were chatting @szhu brought up a good point that we might want to have a broader discussion about what kind of testing we want to include in the frontend.

File: package.json [hunk 1]
```diff
@@ -105,7 +106,47 @@
...
+      "<rootDir>/node_modules/",
+      "<rootDir>/src/database/"
+    ],
+    "transform": {
+      "^.+\\.tsx?$": "ts-jest"
+    },
+    "moduleNameMapper": {
+      ".+\\.(svg|css)$": "<rootDir>/src/__mocks__/svgMock.ts"
+    }
+  },
```
Comment (szhu) [inline package.json:137]: This config is fairly extensive. Can you document where you got some of these snippets from? This should help developers modify it in the future (hint: will help me when adding testing to Gatsby)!

Comment (szhu) [reply package.json:137]: @daschi oops just saw that you merged it. You can just comment it here!

Comment (daschi) [reply package.json:137]: Oops--I'm sorry @szhu I merged before seeing this comment! I mostly looked at [jest](https://jestjs.io/docs/en/getting-started#using-typescript) and [ts-jest](https://kulshekhar.github.io/ts-jest/user/config/) documentation.

**Pull request merged**